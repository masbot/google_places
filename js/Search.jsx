import React, { Component } from 'react';

class Search extends Component {
    handleSearchTermChange = event => {
        this.props.onSearchTermChange(event);
    };

    handlePressEnter = event => {
        this.props.onEnter(event);
    };

    render() {
        return (
            <nav className="navbar navbar-light" style={{ backgroundColor: '#455a64' }}>
                <input
                    style={{ width: '600px' }}
                    type="text"
                    value={this.props.searchTerm}
                    placeholder="Search"
                    onKeyPress={this.handlePressEnter}
                    onChange={this.handleSearchTermChange}
                />
                <div className={'pull-right'} style={{ color: 'white' }}>{this.props.city.name}</div>
            </nav>
        );
    }
}

export default Search;
