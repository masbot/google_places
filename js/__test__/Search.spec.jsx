import React from 'react';
import renderer from 'react-test-renderer';
import Search from '../Search';

test('Search renders correctly', () => {
    const component = renderer.create(<Search city={{ name: '' }} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
