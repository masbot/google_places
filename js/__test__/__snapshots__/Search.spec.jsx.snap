// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<nav
  className="navbar navbar-light"
  style={
    Object {
      "backgroundColor": "#455a64",
    }
  }
>
  <input
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Search"
    style={
      Object {
        "width": "600px",
      }
    }
    type="text"
    value={undefined}
  />
  <div
    className="pull-right"
    style={
      Object {
        "color": "white",
      }
    }
  >
    
  </div>
</nav>
`;
